### ISSUE TITLE
feat: add configurable severity level in createPlugin factory

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#25]
## Problem

Currently, the `createPlugin()` factory function hardcodes `"error"` as the severity level for all rules:

```javascript
const createRuleEntry = (ruleOptions) => ruleOptions ? ["error", ruleOptions] : ["off"];
```

This means users can only choose between `error` or `off` (by passing `false`). To use `warn`, they need to override rules in a separate config object:

```javascript
featureSliced({ /* config */ }),

// Override to warn
{
  rules: {
    '@conarti/feature-sliced/layers-slices': 'warn',
    '@conarti/feature-sliced/public-api': 'warn',
  }
}
```

## Proposed Solution

Add an optional `severity` option to each rule configuration:

```javascript
featureSliced({
  layersSlices: {
    severity: 'warn', // 'error' | 'warn' | 'off'
    allowTypeImports: true,
  },
  publicApi: {
    severity: 'warn',
    level: 'slices',
  },
})
```

Or a global severity option:

```javascript
featureSliced({
  severity: 'warn', // applies to all rules
  layersSlices: { /* options */ },
})
```

## Use Case

This is useful for gradual migration of existing codebases to FSD architecture. Teams can start with `warn` to see violations without breaking CI, then switch to `error` once violations are fixed.

**Comment by conarti** [comment]
Fixed in 2.0.0-rc.16 (afb9e91). You can try it with `npm i -D @conarti/eslint-plugin-feature-sliced@next`; it will ship in 2.0.0.
